enterprise/server/remoteauth/remoteauth.go

```diff
@@ -0,0 +1,182 @@
+package remoteauth
+
+import (
+	"context"
+	"flag"
+	"net/http"
+
+	"github.com/buildbuddy-io/buildbuddy/server/interfaces"
+	"github.com/buildbuddy-io/buildbuddy/server/tables"
+	"github.com/buildbuddy-io/buildbuddy/server/util/authutil"
+	"github.com/buildbuddy-io/buildbuddy/server/util/claims"
+	"github.com/buildbuddy-io/buildbuddy/server/util/grpc_client"
+	"github.com/buildbuddy-io/buildbuddy/server/util/hash"
+	"github.com/buildbuddy-io/buildbuddy/server/util/lru"
+	"github.com/buildbuddy-io/buildbuddy/server/util/status"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+
+	authpb "github.com/buildbuddy-io/buildbuddy/proto/auth"
+)
+
+const (
+	jwtCacheSize = 10_000
```

Comment:
can you add a comment here so that if someone needed to change this they would have enough information to do so?

Reply:
Good call, done.